pkg/cloud/services/eks/tags.go

```diff
@@ -73,6 +76,35 @@ func (s *NodegroupService) reconcileTags(ng *eks.Nodegroup) error {
 	return updateTags(s.EKSClient, ng.NodegroupArn, aws.StringValueMap(ng.Tags), tags)
 }
 
+func (s *NodegroupService) reconcileAsgTags(ng *eks.Nodegroup) error {
+	if s.scope.ManagedMachinePool.Spec.AdditionalAsgTags == nil {
+		return nil
+	}
+	trueVal := true
+	asgType := "auto-scaling-group"
+	for _, asg := range ng.Resources.AutoScalingGroups {
+		input := &autoscaling.CreateOrUpdateTagsInput{}
+		for k, v := range s.scope.ManagedMachinePool.Spec.AdditionalAsgTags {
+			// The k/vCopy is used to address the "Implicit memory aliasing in for loop" issue
+			// https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop
+			kCopy := k
```

Comment:
Can we index the ranged map as given in the stackoverflow link?
```
for key := range s.scope.ManagedMachinePool.Spec.AdditionalAsgTags {
    key:= key (Can remove it and check if it still throws error)
    input.Tags = append(input.Tags, &autoscaling.Tag{
		Key:               &key,
		PropagateAtLaunch: &trueVal,
		ResourceId:        asg.Name,
		ResourceType:      &asgType,
		Value:             & s.scope.ManagedMachinePool.Spec.AdditionalAsgTags[key],
	})
}
if _, err := s.AutoscalingClient.CreateOrUpdateTags(input); err != nil {
	return errors.Wrap(err, "failed to reconcile AutoScalingGroup tags for nodegroup")
}
```

Reply:
I tried to change to this but got error `invalid operation: cannot take address of s.scope.ManagedMachinePool.Spec.AdditionalAsgTags[key] (map index expression of type string)`, so I'll keep it as it for now